sycl/include/CL/sycl/enums.hpp

```diff
@@ -0,0 +1,29 @@
+//==-------------- enums.hpp --- SYCL enums --------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+enum class memory_scope : unsigned char {
```

Comment:
`enum.hpp` sounds too generic. Probably this header should be named `memory_scope.hpp`.

NOTE: there are other header file with similar enum declaration - sycl/include/CL/sycl/ONEAPI/atomic_enums.hpp

NOTE #2: similar header files with enum declarations you can use for reference - sycl/include/CL/sycl/aspects.hpp, sycl/include/CL/sycl/access/access.hpp.

Reply:
Renamed to `memory_enums.hpp` and switched to `using`.